eng/pipelines/common/templates/build-test-job.yml

```diff
@@ -98,6 +105,15 @@ jobs:
           artifactName: '$(librariesBuildArtifactName)'
           displayName: 'live-built libraries'
 
+    - ${{ if eq(parameters.runtimeFlavor, 'mono') }}:  
+      # We need to explictly download CoreCLR for Mono
```

Comment:
Is there tracking issue for that, seems quite unnecessary to me?

Reply:
The issue is at that the CoreCLR tests require it. I'm not sure if there is an issue to remove that dependency;  @trylek  Do you know?

Reply:
There is a somewhat related issue currently assigned to me for moving the tests out of the coreclr subfolder to highlight they are used for both runtimes. I'm not sure what remaining CoreCLR artifacts Mono tests need but it would be certainly reasonable to remove this weird dependency as part of the task.